src/Files.App/Views/MainPage.xaml.cs

```diff
@@ -32,26 +33,19 @@
 
 namespace Files.App.Views
 {
-	/// <summary>
-	/// The root page of Files
-	/// </summary>
 	public sealed partial class MainPage : Page, INotifyPropertyChanged
 	{
-		private VirtualKeyModifiers currentModifiers = VirtualKeyModifiers.None;
-
 		public IUserSettingsService UserSettingsService { get; } = Ioc.Default.GetRequiredService<IUserSettingsService>();
+
 		public ICommandManager Commands { get; } = Ioc.Default.GetRequiredService<ICommandManager>();
+
 		public IWindowContext WindowContext { get; } = Ioc.Default.GetRequiredService<IWindowContext>();
 
 		public SidebarViewModel SidebarAdaptiveViewModel = Ioc.Default.GetRequiredService<SidebarViewModel>();
 
-		public AppModel AppModel => App.AppModel;
+		public static AppModel AppModel => App.AppModel;
 
-		public MainPageViewModel ViewModel
-		{
-			get => (MainPageViewModel)DataContext;
-			set => DataContext = value;
-		}
+		public MainPageViewModel ViewModel { get; } = Ioc.Default.GetRequiredService<MainPageViewModel>();
```

Comment:
Dependency injection should be used here as well

Reply:
I did already. What do you mean? You mean we should do this in the constructor?

Reply:
The preferred way is to register the view model as a singleton in app.xaml.cs and assign it through the constructor.

Reply:
As a lot of code was written like this, I thought this way was preferred. Doesn't matter, I'll fix.

Reply:
See #11820 for reference.

Reply:
> As a lot of code was written like this, I thought this way was preferred. Doesn't matter, I'll fix.

It's a very recent change 👍

Reply:
Fixed.